frontend/src/pages/NewRun.tsx

```diff
@@ -813,6 +847,17 @@ export class NewRun extends Page<{ namespace?: string }, NewRunState> {
         );
         parameters = pipelineVersion.parameters || [];
       }
+      // To avoid breaking current v1 behavior, only allow switch between v1 and v2 when V2 feature is enabled.
+      if (isFeatureEnabled(FeatureKey.V2_ALPHA) && pipeline.id) {
```

Comment:
It won't break v1 behavior directly, but it will break the original v1 test. Because we add a callback logic here, once the pipeline or pipeline version is changed, it will redirect back to the NewRunSwitcher, which makes the following lines of code (this.setStateSafe ...) not be executed at all, and it will fail the existing v1 test.

The green part won't be executed in NewRun.tsx, which lead the failure of the test.

<img width="1728" alt="Screen Shot 2022-08-30 at 5 33 57 PM" src="https://user-images.githubusercontent.com/56132941/187567606-52057f02-4179-4ead-b738-4c0ad537789b.png">

Reply:
I suggest to render <NewRunSwitcher> instead of <NewRun> for the test you mentioned. Hope that completing the state flow with parent component can help resolving the test failure. As a result, we are simplifying the current V1 New Run logic, which avoids potential failure that we might miss.